### PR TITLE
FaucetToken.allocate is now metered

### PR DESCRIPTION
**Pull request by whenceforth**
It will only allocate a pre-determined amount.  We set the amounts during deployment, but we can modify them later if desired. 

The contract owner can access the old allocation behavior via function `allocateTo`.

### CODE REVIEW

File: contracts/tokens/FaucetToken.sol [hunk 1]
```diff
@@ -1,27 +1,68 @@
...
+
+		return doAllocation(msg.sender, perRequestTokenAmount);
+	}
+
+	/**
+  	  * @notice Arbitrarily adds tokens to account
+      * @dev This is for automated testing and for convenience on the alpha test net
+      * @param recipient Account to add tokens to.
+      * @param value Amount to add
+    */
```
Comment (torreyatcitty) [inline contracts/tokens/FaucetToken.sol]: Can you adjust this comment, it's indented all funky.

File: contracts/tokens/FaucetToken.sol [hunk 2]
```diff
@@ -1,27 +1,67 @@
...
-  * @notice A simple token that lets anyone get more of it.
+  * @notice A simple token that lets anyone get more of it, with a cap on the amount that can be allocated per request
   */
-contract FaucetToken is StandardToken {
+contract FaucetToken is StandardToken, Owned {
 
-    string constant public name = "Pig Token";
-    string constant public symbol = "PIG";
-    uint8 constant public decimals = 16;
+	string constant public name = "Pig Token";
```
Comment (hayesgm) [inline contracts/tokens/FaucetToken.sol]: Can we change to spaces 4 as we use elsewhere.

File: contracts/tokens/FaucetToken.sol [hunk 3]
```diff
@@ -1,27 +1,67 @@
...
+    */
+	function allocateTo(address recipient, uint256 value) public returns (bool){
+
+		if(!checkOwner()) {
+			return false;
+		}
+		return doAllocation(recipient, value);
+	}
+
+	function doAllocation(address _owner, uint256 value) internal returns (bool) {
```
Comment (hayesgm) [inline contracts/tokens/FaucetToken.sol]: Can we add a natspec anyway?

File: test/FaucetToken.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
+"use strict";
+
+const BATToken = artifacts.require("./tokens/faucet_tokens/FaucetTokenBAT.sol");
+const OMGToken = artifacts.require("./tokens/faucet_tokens/FaucetTokenOMG.sol");
+
+const utils = require('./utils');
+
+contract('FaucetTokenBAT', function(accounts) {
+    var omgToken;
```
Comment (hayesgm) [inline test/FaucetToken.js]: Can we use spaces 2 for Javascript?

File: test/FaucetToken.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        it("should disburse nothing for a non-configured token", async () => {
+
+            // In addition to attempting an allocation that should do nothing, let's do a
+            // non-limited allocateTo so we can wait for a non-zero balance.
+            const allocatedAmount = 1e18;
+            await(batToken.allocateTo(web3.eth.accounts[1], allocatedAmount));
+
+            await(utils.assertGracefulFailure(batToken, "FaucetToken::AssetNotDisbursable", [], async () => {
+                await(batToken.allocate({from: web3.eth.accounts[1]}));
+                })
```
Comment (hayesgm) [inline test/FaucetToken.js]: semi-colon

**Pull request merged**